src/auth/awsrequestsigner.ts

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GaxiosOptions} from 'gaxios';
+
+import {Headers} from './oauth2client';
+import {Crypto, createCrypto, fromArrayBufferToHex} from '../crypto/crypto';
+
+type HttpMethod =
```

Comment:
Curious if we use a similar type anywhere else in the codebase, or if `gaxios` exposes it? If not, no objection to having it here.

Reply:
It seems that `gaxios` [does not export](https://github.com/googleapis/gaxios/blob/5b586d52b1798b6edf2bec7e59402e4d4115c8ee/src/common.ts#L69) a dedicated type for this.

I couldn't find any similar type in the current repo.